dallinger/command_line/utils.py

```diff
@@ -303,6 +303,18 @@ def verify_id(ctx, param, app):
     return app
 
 
+def run_pre_launch_checks(config):
+    from dallinger.recruiters import ProlificRecruiter, by_name
```

Comment:
I have a couple worries here:
1. we aspire towards a plug-in architecture, where generic Dallinger infrastructure is not aware of the specific details for recruiters (we used to have a lot of this type of check sprinkled throughout Dallinger, and it was a hard battle to get rid of them). Instead of doing the `isinstance()` check here, I think it would be better to introduce a `validate_config(config)` method to the Recruiter interface, and call it on all configured recruiter[s]* (see more about why this is plural in item 2)
2. at one time, it was important to support multiple recruiters for a single experiment. If we're still doing this, we need to make sure we aren't actually dealing with a list of recruiters, and if we are, we need to ask each of them to `validate_config(config)`.

I do think a config validation is a good idea. I bet we could even benefit from adding a real implementation for MTurkRecruiter and others in the future.

Reply:
I think @jessesnyder is totally right.... `isinstance` checks are normally an antipattern! I think it would make a lot of sense to implement `Recruiter.validate_config`. I don't think it would be hard to implement a `MultiRecruiter.validate_config` that iterates over the selected recruiters calling their `validate_config` methods.

Reply:
+1 on the implementation in `MultiRecruiter`. I'd forgotten about that.

Reply:
Thanks a lot @jessesnyder, for your suggestion to add a `Recruiter.validate_config` method. This greatly improves code quality. :+1: